Improve Vault Connect CA docs (#12743)
hashicorp/consul: website/content/docs/connect/ca/vault.mdx

    default mount path `/var/run/secrets/kubernetes.io/serviceaccount/token` if the `jwt` parameter
    is not provided.

> 

Author: This content was all moved up into "Vault Resource Locations" subsection

- `TLSSkipVerify` / `tls_skip_verify` (`bool: false`) - Specifies if SSL peer
  validation should be enforced.

> - `Namespace` / `namespace` (`string: <optional>`) - The Vault Namespace that

Author: This content was moved up into "Vault Resource Locations" subsection

</CodeBlockConfig>

## Configuration with Vault Namespaces

[Vault Namespaces](https://www.vaultproject.io/docs/enterprise/namespaces)
are a multi-tenancy feature of Vault Enterprise that allow
multiple, isolated Vault environments within a single Vault infrastructure.

This Connect CA provider can access Vault tokens, auth methods, and PKI secrets engines
that exist within the same or different Vault Namespaces.

The [`Namespace`](#namespace) option is used as the `X-Vault-Namespace` header
for all API requests made by this CA provider to Vault.
All Vault namespaces included in paths are relative to the value of this header.
For more details on how paths to namespaced resources can be specified,
refer to the [Vault Namespace usage documentation](https://www.vaultproject.io/docs/enterprise/namespaces#usage).

The PKI path options ([`RootPKIPath`](#rootpkipath) and [`IntermediatePKIPath`](#intermediatepkipath)
both support including a Vault Namespace in the path.

> However, the Vault authentication options ([`Token`](#token) and [`AuthMethod`](#authmethod))

Author: @markan : my conclusion from reading the [auth method code](https://github.com/hashicorp/consul/pull/11573/files#diff-7bf30611a760296e2e6ffcd7fe955b0eb8317fe0a43c470b8b52d311af4a2515R744) is that namespaces cannot be provided in the path for that configuration. `AuthMethod.MountPath` seems to be injected to API calls with:
```
loginPath = fmt.Sprintf("auth/%s/login", authMethod.MountPath)
```

... when the Vault Namespace would have to come before `auth/`. So, for both Vault tokens and auth methods, the namespace must be provided in the `Namespace` option rather than in the path. I tried to document that clearly in this section.

Reviewer: You are correct. Also we should note that the Namespace field doesn't affect this in 1.11.0-1.11.5 because of a bug. This is has been corrected in 1.12.0 and backported to the 1.11.x branch (https://github.com/hashicorp/consul/pull/12793). No change is needed, noting this mostly for posterity.

Author: I'll probably include this caveat in the 1.11 version of the docs (backport to that).

I need to rewrite these docs anyways given what we're learned / the config UX enhancements we're in the process of making in 1.12.x for Vault namespaces.

do not allow specifying a Vault Namespace in their configuration.

**If using an authentication mechanism that exists within a non-root Vault Namespace**:
- set the [`Namespace`](#namespace) option to the Vault authentication mechanism's Namespace, and
- `RootPKIPath` and `IntermediatePath` must be in that same Namespace or a child Namespace;
  the relative path to a child Namespace must be included in the path

### Example 1: PKI Paths in Same Namespace

> Consider the following setup of Vault resources and their namespaces:

Reviewer 2: ```suggestion
The following example demonstrates how to configure Consul service mesh to use Vault as the CA when the Vault PKI paths are in the same namespace.

#### Vault Setup
```

| Vault Resource                       | Belongs to Vault Namespace | Resource Name              |
| ------------------------------------ | -------------------------- | -------------------------- |
| `Token`                              | root                       | N/A                        |
| `RootPKIPath` (Primary CA)           | `ns-consul`                | `connect-root`             |
| `IntermediatePKIPath` (Secondary CA) | `ns-consul`                | `connect-intermediate-dc1` |

> The correct Connect CA configuration for this setup is:

Reviewer 2: ```suggestion
#### Consul Service Mesh Configuration

The following `connect` configuration configures Consul service mesh to use Vault as the CA: 
```
I had started adding suggestions to change the language a bit, but I think we can streamline these examples by adding a more descriptive introduction to the examples and using headings to indicate what is happening in each subsection.

<CodeTabs heading="Connect CA configuration" tabs={["Agent configuration", "API"]}>

<CodeBlockConfig filename="/etc/consul.d/config.hcl" highlight="7-9">

```hcl
# ...
connect {
  enabled = true
  ca_provider = "vault"
  ca_config {
    address = "http://localhost:8200"
    token = "<Vault token>"
    root_pki_path = "ns-consul/connect-root"
    intermediate_pki_path = "ns-consul/connect-intermediate-dc1"
  }
}
```

</CodeBlockConfig>

<CodeBlockConfig highlight="5-7">

```json
{
  "Provider": "vault",
  "Config": {
    "Address": "http://localhost:8200",
    "Token": "<Vault token>",
    "RootPKIPath": "ns-consul/connect-root",
    "IntermediatePKIPath": "ns-consul/connect-intermediate-dc1"
  }
}
```

</CodeBlockConfig>

</CodeTabs>

> -> **Why the Namespace option isn't used**:
>   In this case, the [`Namespace`](#namespace) option cannot be set to `ns-consul/` because
>   the Vault `Token` is in the root Vault Namespace.

Reviewer 2: ```suggestion
The [`Namespace`](#namespace) option is not included in this case because the Vault `Token` is in the root Vault namespace. As a result, the Consul `Namespace` cannot be set to `ns-consul/`. 
```
I think this is directly related and probably doesn't need a callout box

### Example 2: Token in Vault Namespace, PKI Paths in Different Namespaces

> Consider the following setup of Vault resources and their namespaces:

Reviewer 2: See previous comment about streamlining these examples.
